torch/distributions/bernoulli.py

```diff
@@ -31,16 +31,22 @@ def __init__(self, probs=None, logits=None):
             raise ValueError("Either `probs` or `logits` must be specified, but not both.")
         if probs is not None:
             self.probs, = broadcast_all(probs)
-            self.logits = probs_to_logits(self.probs, is_binary=True)
         else:
             self.logits, = broadcast_all(logits)
-            self.probs = logits_to_probs(self.logits, is_binary=True)
-        if isinstance(probs, Number):
+        if isinstance(probs, Number) or isinstance(logits, Number):
```

Comment:
One suggestion (feel free to ignore that if you don't want to implement it), for both here and in `Categorical` would be to define `probs_or_logits` in the `if` above, to make it point to whatever was specified to the distribution. You're only using the shape so that's enough, it will save you all the conditionals, and we will be sure that you're not triggering the lazy attributes by accident.

Reply:
Good idea..let me refactor this part.